Amethyst/Managers/ScreenManager.swift

```diff
@@ -278,12 +279,21 @@ final class ScreenManager<Delegate: ScreenManagerDelegate>: NSObject, Codable {
     }
 
     func selectLayout(_ layoutString: String) {
-        guard let layoutIndex = layouts.firstIndex(where: { $0.layoutKey == layoutString }) else {
+        log.debug("layout is " + layoutString)
+
+        guard let currentLayout = currentLayout?.layoutKey else {
+            return
+        }
+
+        let nextLayout = currentLayout == layoutString ? previousLayout : layoutString
+
+        guard let layoutIndex = layouts.firstIndex(where: { $0.layoutKey == nextLayout }) else {
```

Comment:
I think there is a `first` function for getting the layout itself instead of getting the index.

Reply:
Can you explain some more what you mean here please? 

I did not change that part of the code and this function `setCurrentLayoutIndex(layoutIndex)` depends on the layoutIndex.

Reply:
Ah, you're right. Yeah, let's keep the existing code.